distro/wildfly/subsystem/src/main/java/org/camunda/bpm/container/impl/jboss/service/MscManagedProcessEngineController.java

```diff
@@ -255,33 +263,29 @@ public InjectedValue<MscRuntimeContainerJobExecutor> getMscRuntimeContainerJobEx
     return mscRuntimeContainerJobExecutorInjector;
   }
 
-  public static void initializeServiceBuilder(ManagedProcessEngineMetadata processEngineConfiguration, MscManagedProcessEngineController service,
-          ServiceBuilder<ProcessEngine> serviceBuilder, String jobExecutorName) {
+  public void initializeServiceBuilder(ManagedProcessEngineMetadata processEngineConfiguration,
+      ServiceBuilder<ProcessEngine> serviceBuilder, String jobExecutorName) {
 
     ContextNames.BindInfo datasourceBindInfo = ContextNames.bindInfoFor(processEngineConfiguration.getDatasourceJndiName());
-    serviceBuilder.addDependency(ServiceName.JBOSS.append("txn").append("TransactionManager"), TransactionManager.class, service.getTransactionManagerInjector())
-      .addDependency(datasourceBindInfo.getBinderServiceName(), DataSourceReferenceFactoryService.class, service.getDatasourceBinderServiceInjector())
-      .addDependency(ServiceNames.forMscRuntimeContainerDelegate(), MscRuntimeContainerDelegate.class, service.getRuntimeContainerDelegateInjector())
-      .addDependency(ServiceNames.forMscRuntimeContainerJobExecutorService(jobExecutorName), MscRuntimeContainerJobExecutor.class, service.getMscRuntimeContainerJobExecutorInjector())
-      .addDependency(ServiceNames.forMscExecutorService())
+    serviceBuilder.addDependency(ServiceName.JBOSS.append("txn").append("TransactionManager"), TransactionManager.class, transactionManagerInjector)
+      .addDependency(datasourceBindInfo.getBinderServiceName(), DataSourceReferenceFactoryService.class, datasourceBinderServiceInjector)
+      .addDependency(ServiceNames.forMscRuntimeContainerDelegate(), MscRuntimeContainerDelegate.class, runtimeContainerDelegateInjector)
+      .addDependency(ServiceNames.forMscRuntimeContainerJobExecutorService(jobExecutorName), MscRuntimeContainerJobExecutor.class, mscRuntimeContainerJobExecutorInjector)
```

Comment:
I focused on changing the code that was broken for WildFly 27. The `addDependency` API still works, also in combination with the supplier-driven API.
Cleaning up other occurrences and removing deprecated (but not removed) API calls will be part of #3589.

Reply:
That sounds reasonable in general. But then I don't fully understand yet why we moved to `requires` calls in the  `ProcessApplicationDeploymentProcessor`. I struggle to make the connection between breaking changes and moving away from `addDependency` in that class since this accounts for ~80% of all changes in this PR.

I just want to make sure we understand what changes are required and what changes are optional (and can be solved with https://github.com/camunda/camunda-bpm-platform/issues/3589). The current behavior Yana and Tassilo are investigating might potentially be related to those changes.

Reply:
After discussion with Tassilo and Miklas, we **decided** to **revert** the made changes for `#addDependencies` (and the Injectors) so that we don't implement a workaround for stop service and the missing process application.
**Argument**: the risk is higher with the workaround and we don't fully see the complete picture yet. (secondary: code will be more consistent; we can update to the new API later)